tests/trinity/integration/test_boot.py

```diff
@@ -71,6 +71,25 @@ def async_process_runner():
     })
 
 
+@pytest.mark.parametrize(
+    'command',
+    (
+        ['trinity', '--light', '--tx-pool'],
+        ['trinity', '--light', '--ropsten', '--tx-pool'],
+    )
+)
+@pytest.mark.asyncio
+async def test_txpool_deactivated(async_process_runner, command):
+    # UPNP discovery can delay things, we use a timeout longer than the discovery timeout
+    await async_process_runner.run(command, timeout_sec=40)
```

Comment:
Can't we disable UPNP altogether for this test? I also wonder if an integration test is the right approach to ensure the tx pool is deactivated...

Reply:
>Can't we disable UPNP altogether for this test

Oh, indeed we don't have to wait for UPNP discovery here. Will update.

>I also wonder if an integration test is the right approach to ensure the tx pool is deactivated

I have mixed feelings about this test myself. After all, I guess the main thing that we want to achieve is to ensure someone running `trinity --light --tx-pool` won't be left with something totally broken which is what the *exception scanning* test tries to achieve.

But then I think it doesn't hurt to go one step beyond that and test that the tx pool doesn't start at all and yields a warning. I think it's hard to test this somewhere else without making the test somewhat artificial and redundant. Testing this on the integration level gives us a 100 % guarantee and only adds up a couple of seconds.

Not sure. How does the rest feel about it? @pipermerriam @carver